src/common/meta/src/key/node_address.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+use snafu::OptionExt;
+
+use crate::error::{InvalidMetadataSnafu, Result};
+use crate::key::{MetadataKey, NODE_ADDRESS_PATTERN, NODE_ADDRESS_PREFIX};
+use crate::peer::Peer;
+
+/// The key stores node address.
+///
+/// The layout: `__node_address/{node_id}`
+#[derive(Debug, PartialEq)]
+pub struct NodeAddressKey {
+    pub node_id: u64,
+}
```

Comment:
We need to distinguish between Flownode and Datanode 🥲

Reply:
Datanodes and Flownode should use the same ID generator?

Reply:
cc @zyy17 @daviderli614

Reply:
I have already added the `role` field to `NodeAddressKey`.